snapcraft/storeapi/errors.py

```diff
@@ -379,6 +379,45 @@ def __fmt_error_unknown(self, response):
         return fmt
 
 
+class StoreMetadataError(StoreError):
+
+    __FMT_NOT_REGISTERED = (
+        'Sorry, try `snapcraft register {snap_name}` before trying to '
+        'update metadata.')
+
+    fmt = 'Received {status_code!r}: {text!r}'
+
+    def __init__(self, snap_name, response, metadata):
+        try:
+            response_json = response.json()
+        except (AttributeError, JSONDecodeError):
+            response_json = {}
+
+        if response.status_code == 404:
+            self.fmt = self.__FMT_NOT_REGISTERED
+        elif response.status_code == 409:
+            conflicts = [(error['extra']['name'], error)
+                         for error in response_json['error_list']
+                         if error['code'] == 'conflict']
+            parts = ["Metadata not updated!"]
+            for field_name, error in sorted(conflicts):
+                sent = metadata.get(field_name)
+                parts.extend((
+                    "Conflict in {!r} field:".format(field_name),
+                    "    In snapcraft.yaml: {!r}".format(sent),
+                    "    In the Store:      {!r}".format(error['message']),
+                ))
+            parts.extend((
+                "You can repeat the command with --force-metadata",
```

Comment:
This comma introduces a weird new line in the middle of the message. We generally don't wrap the messages, and if we do, we probably should do it in a single place. So, I think it would be better to add a single string.

Reply:
Also related to this message, "You can repeat the command with --force-metadata" seems confusing when you get this error while pushing a new revision. Note that in that case the revision is correctly uploaded, so re-uploading it won't work (so you would need to do a --only-metadata + --force-metadata instead).

Reply:
Regarding the comma, I was doing that *on purpose*, I'll fix it to have a single string.

And will improve the message.